dataplane/ebpf/src/egress/tcp.rs

```diff
@@ -68,6 +71,39 @@ pub fn handle_tcp_egress(ctx: TcContext) -> Result<i32, i64> {
     unsafe { (*tcp_hdr).check = 0 };
 
     // TODO: connection tracking cleanup https://github.com/kubernetes-sigs/blixt/issues/85
+    // SNAT the port
+    unsafe { (*tcp_hdr).source = tcp_backend.backend_key.port as u16 };
+
+    let tcp_hdr_ref = unsafe { tcp_hdr.as_ref().ok_or(TC_ACT_OK)? };
+
+    // If the packet has the RST flag set, it means the connection is being terminated, so remove it
+    // from our map.
+    if tcp_hdr_ref.rst() == 1 {
+        unsafe {
+            TCP_CONNECTIONS.remove(&client_key)?;
+        }
+    }
+
+    let mut tcp_state = tcp_backend.state;
+    let moved = handle_tcp_conn_next_state(tcp_hdr_ref, &mut tcp_state);
+    // If the connection has moved to the Closed state, stop tracking it.
+    if let TCPState::Closed = tcp_state {
+        unsafe {
+            TCP_CONNECTIONS.remove(&client_key)?;
+        }
+    // If the connection has not reached the Closed state yet, but it did advance to a new state,
+    // then record the new state.
+    } else if moved {
+        let bk = *tcp_backend;
+        let new_tcp_backend = TCPBackend {
+            backend: bk.backend,
+            backend_key: bk.backend_key,
+            state: tcp_state,
+        };
+        unsafe {
+            TCP_CONNECTIONS.insert(&client_key, &new_tcp_backend, 0_u64)?;
+        }
```

Comment:
nit: can you just internalize all of this into `handle_tcp_conn_close` or an equalivalent helper function?